service/history/timerQueueProcessor2_test.go

```diff
@@ -361,6 +368,7 @@ func (s *timerQueueProcessor2Suite) TestWorkflowTimeout_Cron() {
 	schedule := "@every 30s"
 
 	builder := newMutableStateBuilderWithEventV2(s.mockShard, s.mockEventsCache, s.logger, we.GetRunId())
+	s.mockShardManager.On("UpdateShard", mock.Anything).Return(nil).Maybe()
```

Comment:
why is this `Maybe()`. call should either be made or not, otherwise the test isnt really deterministic.

Reply:
This test is not deterministic. I tried to use Once(). But it will fail when running the test as the suite and succeed when running as a single test

Reply:
i think we should figure out why its not deterministic and fix that. otherwise this will come to bite us later..

Reply:
This test is testing async method. So the calls might vary.